ksqldb-engine/src/main/java/io/confluent/ksql/serde/SerdeFeaturesFactory.java

```diff
@@ -87,6 +88,57 @@ public static SerdeFeatures buildValueFeatures(
     return SerdeFeatures.from(builder.build());
   }
 
+  /**
+   * Not all {@code KeyFormat}s are valid internal topic formats. Specifically,
+   * we want to ensure that the key format is (1) not NONE and (2) explicitly sets
+   * the wrapping if it contains only a single column. This method ensures that
+   * both of these are eagerly set.
+   *
+   * @return the key format to use for the output of the aggregation
```

Comment:
This isn't just for aggregations, right? I'm confused by this javadoc.

Reply:
yes you are right. I wrote the javadoc and then I did some refactoring and it no longer applies! good catch